illumina_somaticVariants.pm

```diff
@@ -34,8 +34,13 @@ sub parseSamples {
         if($opt{SET_HAS_METADATA} eq "yes"){
             print "Reading Tumour/Reference names from metadata\n\n";
             my $metadata = metadataParse($opt{OUTPUT_DIR});
-            push(@{$somatic_samples{$cpct_name}{"ref"}}, $metadata->{'ref_sample'});
-            push(@{$somatic_samples{$cpct_name}{"tumor"}}, $metadata->{'tumor_sample'});
+            if($sample =~ $metadata->{'ref_sample'}){
+              push(@{$somatic_samples{$cpct_name}{"ref"}}, $metadata->{'ref_sample'});
+            } elsif ($sample =~ $metadata->{'tumor_sample'}){
+              push(@{$somatic_samples{$cpct_name}{"tumor"}}, $metadata->{'tumor_sample'});
+            } else {
+              print "WARNING: Somatic variants are set to yes but metadata does not match sample names";
```

Comment:
As we'll be the only ones running it, I think it'll be clear enough. The important thing is the functionality, this thing will never be a glaring example of excellent code.